tests/logger.cpp

```diff
@@ -0,0 +1,58 @@
+#include "logger.h"
+#include <gtest/gtest.h>
+#include "hitRecord.h"
+#include "intersectables/meshTriangle.h"
+#include "intersectables/sphere.h"
+#include "intersectables/triangle.h"
+#include "materials/diffuse.h"
+#include "math/matrix3f.h"
+#include "math/matrix4f.h"
+#include "math/vector3f.h"
+#include "math/vector4f.h"
+#include "objReader.h"
+#include "ray.h"
+#include "spectrum.h"
+
+TEST(Logger, log_vector3f) { Logger().log(Vector3f()); }
+
+TEST(Logger, log_vector4f) { Logger().log(Vector4f()); }
+
+TEST(Logger, log_spectrum) { Logger().log(Spectrum(1, 1, 1)); }
+
+TEST(Logger, log_matrix3f) { Logger().log(Matrix3f()); }
+
+TEST(Logger, log_matrix4f) { Logger().log(Matrix3f()); }
+
+TEST(Logger, log_ray) {
+  Ray ray(new Vector3f(), new Vector3f());
+  Logger().log(ray);
```

Comment:
To make this better testable, you could split the method into string conversion/actual logging, i.e:

ToString(MeshTriangle&) {
}

template <typename Loggable>
Log(const Loggable& l) {
  std::out << ToString(l);
}